src/export/StructureDefinitionExporter.ts

```diff
@@ -414,9 +414,9 @@ export class StructureDefinitionExporter implements Fishable {
         // The FHIR spec prohibits constraining any parent element in a 'specialization'
         // (i.e., logical model and resource), therefore log an error if that is attempted
         // and continue to the next rule.
-        if (element.path !== element.base.path) {
-          // The AddElementRule always sets the newElementBase.path to the value of element.path.
-          // All parent elements will have the element.base.path pointing to the parent.
+        const baseElementPath = element.path.replace(`${element.structDef.id}.`, '');
+        const addElementRules = fshDefinition.rules.filter(rule => rule instanceof AddElementRule);
+        if (rule.path && !addElementRules.some(rule => baseElementPath.includes(rule.path))) {
```

Comment:
I think that technically we want to use `startsWith` rather than `includes`, right?  Because we don't want to potentially catch matches in the middle or end of the path (without the beginning of the path).  In fact, to really do this right, I think we would check for an exact match or a startsWith having the path and a trailing `.`.  You might be able to even remove the `baseElementPath` const on line 417 and just do this instead:
```
if (rule.path && !addElementRules.some(rule => {
  return
    element.path === `${element.structDef.type}.${rule.path}`
    || element.path.startsWith(`${element.structDef.type}.${rule.path}.`)
}));
```